Gems/AWSCore/Code/Tests/Credential/AWSDefaultCredentialHandlerTest.cpp

```diff
@@ -100,26 +116,54 @@ TEST_F(AWSDefaultCredentialHandlerTest, GetCredentialsProvider_ProfileCredential
     Aws::Auth::AWSCredentials nonEmptyCredential(AWS_ACCESS_KEY, AWS_SECRET_KEY);
     EXPECT_CALL(*m_environmentCredentialsProviderMock, GetAWSCredentials()).Times(1).WillOnce(::testing::Return(emptyCredential));
     EXPECT_CALL(*m_profileCredentialsProviderMock, GetAWSCredentials()).Times(1).WillOnce(::testing::Return(nonEmptyCredential));
+    EXPECT_CALL(*m_instanceProfileCredentialsProviderMock, GetAWSCredentials()).Times(0);
     auto credentialProvider = m_credentialHandler->GetCredentialsProvider();
     EXPECT_TRUE(credentialProvider == m_profileCredentialsProviderMock);
 }
 
 TEST_F(AWSDefaultCredentialHandlerTest, GetCredentialsProvider_ProfileNameHasBeenChanged_GetDifferentCredentialProvider)
 {
     Aws::Auth::AWSCredentials emptyCredential;
-    Aws::Auth::AWSCredentials nonEmptyCredential(AWS_ACCESS_KEY, AWS_SECRET_KEY);
     EXPECT_CALL(*m_environmentCredentialsProviderMock, GetAWSCredentials()).Times(1).WillOnce(::testing::Return(emptyCredential));
     EXPECT_CALL(*m_profileCredentialsProviderMock, GetAWSCredentials()).Times(0);
+    EXPECT_CALL(*m_instanceProfileCredentialsProviderMock, GetAWSCredentials()).Times(1).WillOnce(::testing::Return(emptyCredential));
     m_profileName = "dummyProfile";
     auto credentialProvider = m_credentialHandler->GetCredentialsProvider();
     EXPECT_TRUE(credentialProvider != m_profileCredentialsProviderMock);
 }
 
+TEST_F(AWSDefaultCredentialHandlerTest, GetCredentialsProvider_InstanceProfileCredentialProviderReturnsNonEmptyCredentials_GetExpectedCredentialsProvider)
+{
+    Aws::Auth::AWSCredentials emptyCredential;
+    Aws::Auth::AWSCredentials nonEmptyCredential(AWS_ACCESS_KEY, AWS_SECRET_KEY);
+    EXPECT_CALL(*m_environmentCredentialsProviderMock, GetAWSCredentials()).Times(1).WillOnce(::testing::Return(emptyCredential));
+    EXPECT_CALL(*m_profileCredentialsProviderMock, GetAWSCredentials()).Times(1).WillOnce(::testing::Return(emptyCredential));
+    EXPECT_CALL(*m_instanceProfileCredentialsProviderMock, GetAWSCredentials()).Times(1).WillOnce(::testing::Return(nonEmptyCredential));
+    auto credentialProvider = m_credentialHandler->GetCredentialsProvider();
+    EXPECT_TRUE(credentialProvider == m_instanceProfileCredentialsProviderMock);
+}
+
+TEST_F(AWSDefaultCredentialHandlerTest, GetCredentialsProvider_InstanceMetadataDisabled_GetDifferentCredentialProvider)
+{
+    // save current value so we can restore it after the test
+    const auto currentEc2MetadataDisabledValue = Aws::Environment::GetEnv(AWS_EC2_METADATA_DISABLED);
+    AZ::Utils::SetEnv(AWS_EC2_METADATA_DISABLED, "truE", 1);
```

Comment:
Is `truE` intended?

Reply:
yes! I know it looks weird, but its to ensure character casing doesn't matter. There is a [similar check](https://github.com/aws/aws-sdk-cpp/blob/main/aws-cpp-sdk-core-tests/aws/auth/AWSCredentialsProviderTest.cpp#L311) in the AWS SDK C++.

Reply:
maybe put this in a named const / variable or use data driven paradigm for test to prove casing doesn't matter, but other values that don't equal value.to_lower() == "true" fail?

Reply:
you're right, better to have an explicit test so its clear what the intent is. Will update.